src/utils.js

```diff
@@ -513,14 +513,21 @@ const utils = {
   align: function (points, line) {
     const tx = line.p1.x,
       ty = line.p1.y,
-      a = -atan2(line.p2.y - ty, line.p2.x - tx),
-      d = function (v) {
-        return {
-          x: (v.x - tx) * cos(a) - (v.y - ty) * sin(a),
-          y: (v.x - tx) * sin(a) + (v.y - ty) * cos(a),
-        };
+      dx = line.p2.x - tx,
+      dy = line.p2.y - ty,
+      magnitude = sqrt(dx ** 2 + dy ** 2),
+      unitX = dx / magnitude,
+      unitY = dy / magnitude;
+
+    return points.map((p) => {
+      const x = p.x - tx;
+      const y = p.y - ty;
+      // Apply a basis change on x,y to align the x axis of the curve with the line
```

Comment:
not sure the comment is 100% accurate
I understand that the curve move to p1, p1 becoming it's zero point and is rotated to align the line axis
is that correct?